pkg/oauth/apiserver/oauth_apiserver.go

```diff
@@ -40,6 +41,9 @@ type OAuthServerConfig struct {
 	// KubeClient is kubeclient with enough permission for the auth API
 	KubeClient kclientset.Interface
 
+	// KubeExternalClient is for creating user events
+	KubeExternalClient kclientsetexternal.Interface
+
```

Comment:
Add an events client directly in this config and build it in `NewOAuthServerConfig`